src/libraries/System.IO.Compression/src/System/IO/Compression/ZipArchiveEntry.cs

```diff
@@ -111,7 +112,8 @@ internal ZipArchiveEntry(ZipArchive archive, string entryName)
             _versionMadeByPlatform = CurrentZipPlatform;
             _versionMadeBySpecification = ZipVersionNeededValues.Default;
             _versionToExtract = ZipVersionNeededValues.Default; // this must happen before following two assignment
-            _generalPurposeBitFlag = 0;
+            _compressionLevel = null;
+            _generalPurposeBitFlag = MapDeflateCompressionOption(0, _compressionLevel);
```

Comment:
Good change. If we ever decide to set an actual value in `_compressionLevel` in the previous line, `_generalPurposeBitFlag` will change appropriately.